orchestrator/workflow.py

```diff
@@ -1393,6 +1404,7 @@ def _exec_steps(steps: StepList, starting_process: Process, dblogstep: StepLogFu
         # as bare exceptions are not JSON serializable
         result_to_log = step_result_process.on_failed(error_state_to_dict).on_waiting(error_state_to_dict)
         result_to_log.on_success(mutationlogger).on_failed(errorlogger).on_waiting(errorlogger)
+        result_to_log.on_failed(broadcast_failed_step)
```

Comment:
Nice. We also need to run at after successful completion of a process though? For when a retried process transitions from failed to successful.

Which makes me think that we might as well call `broadcast_failed_step()` one level up in runwf() after it returns from `_exec_steps(steps, next_state, _logstep)`.

Reply:
I like @tjeerddie suggestion better to do it on the resume endpoint + here on step failure. Then we don't need what I suggested